lib/src/list.dart

```diff
@@ -111,12 +111,23 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
     }
 
     try {
-      final value = realmCore.listGetElementAt(this, index);
-
+      var value = realmCore.listGetElementAt(this, index);
       if (value is RealmObjectHandle) {
-        return realm.createObject(T, value, _metadata!) as T;
+        late RealmObjectMetadata targetMetadata;
+        late Type type;
+        if (T == RealmValue) {
+          final tuple = realm.metadata.getByClassKey(realmCore.getClassKey(value));
+          type = tuple.item1;
+          targetMetadata = tuple.item2;
```

Comment:
I don't think this handles the case where you open a Realm with an incomplete schema that contains `RealmValue` properties linking to tables not in the user schema.

Reply:
We should return dynamic objects.

Reply:
Hmm .. added a test, but it crashes in
```dart
realmCore.getProperty(object, propertyMeta.key);
```
At least the c-api don't support this